### PR TITLE
Update contributing.md

### PR DESCRIPTION
**Pull request by Dridou**
Translation of the complete file, some errors still occure

I'm not sure at all about "you can base your work off of master." traduction line 18
Line 20 I don't know how to translate the examples
Line 21 I'm quite sure "tickets" is not the good term
I still don't know if I should let terms as "issues" "requests" in english or not, depending who is reading, especially for "issue" for example ,any suggestion?
I admit I've been tired at the end of this, that's why some stupid errors might be occured, sorry for that. But I hope this (big) translation will help you!

### CODE REVIEW

**Comment by Dridou** [comment]
[UP] I just saw that I also forgot to delete a complete paragraph which is translated.

**Comment by anonimal** [comment]
>Line 21 I'm quite sure "tickets" is not the good term

This is a remanent of Trac and is usually well-known throughout developer communities.

Let's see if @MoroccanMalinois would be willing to help with the remaining translation questions 😄

**Comment by Josexv1** [comment]
I think most of the tech wording needs to be in English, like Issue, Issue Tracker, daemon, commit, or something in that category as most of us developers are aware of it and feel more comfortable using the mainstream keywords.

I have translated the Examples in Spanish, but i guess it can be in English, since all the work needs to be in English.

**Comment by Dridou** [comment]
I think you're right, I really hesitated though, since it's my first translation that's cool to have other's mind about it

File: kovri.i2p/_i18n/fr/contributing.md [hunk 1]
```diff
@@ -78,76 +79,96 @@ C4 is meant to provide a reusable optimal collaboration model for open source so
...
 - A new Contributor who makes a correct patch SHALL be invited to become a Maintainer.
 - Administrators MAY remove Maintainers who are inactive for an extended period of time, or who repeatedly fail to apply this process accurately.
 - Administrators SHOULD block or ban "bad actors" who cause stress and pain to others in the project. This should be done after public discussion, with a chance for all parties to speak. A bad actor is someone who repeatedly ignores the rules and culture of the project, who is needlessly argumentative or hostile, or who is offensive, and who is unable to self-correct their behavior when asked to do so by others.
 
-# Governance Process
+- Les fondateurs du projet SE COMPORTENT comme des Admin pour gérer l'ensemble des Mainteneurs du projet.
+- Les Admins ASSURENT leur propre succession en promouvant les Mainteneurs les plus efficaces.
+- Un nouveau contributeur qui fait un patch correct EST INVITÉ à devenir Mainteneur.
+- Les admins sont priés de supprimer les Mainteneurs qui sont inactifs pendant une période trop longue, ou n'appliquent pas le processus répétivement.
```
Comment (MoroccanMalinois) [inline kovri.i2p/_i18n/fr/contributing.md]: > Les admins sont priés de

Les administrateurs PEUVENT

File: kovri.i2p/_i18n/fr/contributing.md [hunk 2]
```diff
@@ -78,76 +79,96 @@ C4 is meant to provide a reusable optimal collaboration model for open source so
...
+- Les admins doivent blocker ou bannir "les mauvais acteurs" qui causent du stress et de la douleur aux autres dans le projet. Cela doit être fait après une discussion génerale, avec une chance pour tous les partis de parler. Un mauvais acteur est quelqu'un qui ignore plusieurs fois les règlent and la culture du projet, qui est sans arguments ou hostile, ou qui est offensant, and qui est incapable de se comporter de façon correct quand les autres lui demandent.
```
Comment (MoroccanMalinois) [inline kovri.i2p/_i18n/fr/contributing.md]: > doivent

DEVRONT

> les règlent and

les règles et

> and qui

et qui

File: kovri.i2p/_i18n/fr/contributing.md [hunk 3]
```diff
@@ -1,72 +1,73 @@
...
+3. Creez une [branche pour le sujet concerné](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
+4. [**Signez**](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) votre ou vos commit(s)
+5. Envoyez une pull-request à la branche ```master```
+   - Nous n'avons pas encore de tags du au fait que nous soyons en pré-alpha. Donc pour l'instant vous pouvez directement travailler dans la branche master.
+   - Les messages de commmit doivent être détaillés par défaut, accompagné d'un message court d'environ 50 caractères à propos du sujet, une ligne blanche, et d'un texte détaillé expliquatoire constitué d'un ou plusieurs paragraphe(s) séparés - à moins que le titre explique à lui même le commit.
+   - Le titre du commit doit préciser la class ou l'aspect du projet concerné. Par exemple, "HTTPProxy: implement User-Agent scrubber. Fixes #193." ou "Garlic: fix uninitialized padding in ElGamalBlock".
+   - Si un commit particulier se réfère à un autre problème, veuillez dans ce cas ajouter une réference. Par exemple "Voir #123", ou "#123 résolu". Cela nous aidera à résoudre les tickets quand nous mergerons dans ```master```.
```
Comment (Josexv1) [inline kovri.i2p/_i18n/fr/contributing.md]: We may need to specify that everything must me commented/typed in English, we may end up with multiple language PR and will be hard to handle.

Comment (anonimal) [inline kovri.i2p/_i18n/fr/contributing.md]: >Are commit messages in other languages accepted ? variables/functions names ? Feels like this should be specified

>imho, these examples should not be translated, as commit messages must be in engish.

>We may need to specify that everything must me commented/typed in English, we may end up with multiple language PR and will be hard to handle.

@MoroccanMalinois @Josexv1 I completely agree: English everywhere when possible. I'm not necessarily favoring the English language but rather favoring consistency and the fact that English is taught basically all over the globe. As such, I think we should stick with English.

File: kovri.i2p/_i18n/fr/contributing.md [hunk 4]
```diff
@@ -1,72 +1,73 @@
...
+
+## Envoyer votre travail
+Pour mettre à contribution votre travail, veuillez procéder de la façon suivante:
 
-## Proposals
-To contribute a proposal, please review our [open issues](https://github.com/monero-project/kovri/issues) for existing proposals. If what you propose is not there, then [open a new issue](https://github.com/monero-project/kovri/issues/new).
+1. Fork Kovri
+2. Lisez notre [style guide](https://github.com/monero-project/kovri/blob/master/doc/STYLE.md)
+3. Creez une [branche pour le sujet concerné](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging)
+4. [**Signez**](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work) votre ou vos commit(s)
```
Comment (Dridou) [inline kovri.i2p/_i18n/fr/contributing.md]: You also translate links? I didn't know if that could work in French too.

File: kovri.i2p/_i18n/fr/contributing.md [hunk 5]
```diff
@@ -78,76 +79,96 @@ C4 is meant to provide a reusable optimal collaboration model for open source so
 - Maintainers SHALL have commit access to the repository.
 - Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor under the terms of this contract.
 
-### Licensing and Ownership
+- Le projet UTILISE le système de révision de contrôle fourni par GIT.
```
Comment (Dridou) [inline kovri.i2p/_i18n/fr/contributing.md]: For this change and next ones, I hesitated but this "should" suggests  that it is used, that's why here and all the following examples, I translated in that way and not in your, I trust you if you think it's better though.

File: kovri.i2p/_i18n/fr/contributing.md [hunk 6]
```diff
@@ -78,76 +79,96 @@ C4 is meant to provide a reusable optimal collaboration model for open source so
...
+
+- Les changements du projet SONT GOURVERNÉS par le principe d'identifier précisement les problèmes et d'y appliquer des solutions minimales et précises.
+- Pour proposer des changements, un utilisateur doit créer une issue dans le tracker de problèmes du projet Plateforme.
+- L'utilisateur ou le Contributeur doit écrire l'issue en décrivant le problème observé.
+- L'utilisateur ou le Contributeur doit estimer la précision de leur observation et la valeur de la résolution de leur problème.
+- Les utilisateurs ne CRÉENT PAS de requêtes de fonctionnalités, idées, suggestions ou quelconque solutions aux problèmes qui ne sont pas explicitement documentés et founis.
+- Ainsi, la sortie finale du projet EST une liste de problèmes observés puis fixés.
+- Pour travailler sur un problème, un Contributeur FORK le répertoire du projet and travaille ensuite sur leur répertoire fork.
+- Pour soumettre un patch, un Contributeur CRÉE une requête Plateforme pour le projet.
+- Un contributeur ne COMMIT PAS directement les changements sur le projet.
```
Comment (Dridou) [inline kovri.i2p/_i18n/fr/contributing.md]: "commiter" or "commit" ? because @anonimal specified to let in english as long as  we can

**Comment by Dridou** [comment]
Thank you very much @MoroccanMalinois for corrections which, that is reassuring me, are most of the time due to the tiredness or inattentiveness, maybe I should translate in several times, not all at once.

File: kovri.i2p/_i18n/fr/contributing.md (hunk 4 above)
Comment (MoroccanMalinois) [inline kovri.i2p/_i18n/fr/contributing.md]: It's not about translating the link. It just turns out that git have a french translation of this exact doc.

File: kovri.i2p/_i18n/fr/contributing.md (hunk 5 above)
Comment (MoroccanMalinois) [inline kovri.i2p/_i18n/fr/contributing.md]: I used [this translation of RFC 2119]( http://abcdrfc.free.fr/rfc-vf/rtf/rfc2119.rtf). Those are codified words and imho, we don't have a choice.

File: kovri.i2p/_i18n/fr/contributing.md (hunk 6 above)
Comment (MoroccanMalinois) [inline kovri.i2p/_i18n/fr/contributing.md]: Correct :+1:

**Comment by MoroccanMalinois** [comment]
Thanks @Dridou . Yeah it was a long one and C4 is full of nuances. Hopefully, the rest should be easier.

**Comment by anonimal** [comment]
@Dridou all translations are now done in the [kovri-docs repo](https://github.com/monero-project/kovri-docs). You'll have to fork that repo, make the changes, and submit a new PR instead of rebasing this PR.

**Comment by Dridou** [comment]
Okay sure, I'll do that in the next days

Adrien Nebon-Carle
Tél: 06 85 85 24 01
Étudiant EPITA École d'ingénieur en informatique

Le 5 août 2017 à 23:51, 0x914409F1 <notifications@github.com<mailto:notifications@github.com>> a écrit :


@Dridou<https://github.com/dridou> all translations are now done in the kovri-docs repo<https://github.com/monero-project/kovri-docs>. You'll have to fork that repo, make the changes, and submit a new PR instead of rebasing this PR.

**Comment by anonimal** [comment]
Resolved in https://github.com/monero-project/kovri-docs/pull/25. Thank you @Dridou for your contribution!

**Comment by anonimal** [comment]
(also, thanks to @MoroccanMalinois and @Josexv1 for the reviews)

**Pull request closed without merging**